proc/proc.go

```diff
@@ -146,69 +145,50 @@ func (dbp *Process) LoadInformation(path string) error {
 	return nil
 }
 
-// Find a location by string (file+line, function, breakpoint id, addr)
-func (dbp *Process) FindLocation(str string) (uint64, error) {
-	// File + Line
-	if strings.ContainsRune(str, ':') {
-		fl := strings.Split(str, ":")
+func (dbp *Process) FindFileLocation(fileName string, lineno int) (uint64, error) {
+	pc, _, err := dbp.goSymTable.LineToPC(fileName, lineno)
+	if err != nil {
+		return 0, err
+	}
+	return pc, nil
+}
 
-		fileName, err := filepath.Abs(fl[0])
-		if err != nil {
-			return 0, err
-		}
+// Finds address of a function's line
+// If firstLine == true is passed FindFunctionLocation will attempt to find the first line of the function
+// If lineOffset is passed FindFunctionLocation will return the address of that line
+// Pass lineOffset == 0 and firstLine == false if you want the address for the function's entry point
+// note that setting breakpoints at that address will cause surpirsing behavior:
+// https://github.com/derekparker/delve/issues/170
+func (dbp *Process) FindFunctionLocation(funcName string, firstLine bool, lineOffset int) (uint64, error) {
+	fn := dbp.goSymTable.LookupFunc(funcName)
+	if fn == nil {
+		return 0, fmt.Errorf("Could not find function %s\n", funcName)
+	}
 
-		line, err := strconv.Atoi(fl[1])
-		if err != nil {
-			return 0, err
+	if firstLine {
+		filename, lineno, _ := dbp.goSymTable.PCToLine(fn.Entry)
+		if filepath.Ext(filename) != ".go" {
+			return fn.Entry, nil
 		}
 
-		pc, _, err := dbp.goSymTable.LineToPC(fileName, line)
+		lines, err := dbp.ast.NextLines(filename, lineno)
```

Comment:
More thoughts: A non-Go function will obviously not have the stack check we're trying to avoid here in function entry. We could either:

a) Pass `filepath.Base(filename) == ".go"` for firstLine, so that we skip this branch altogether for non-Go files.
b) Use a `NextLines` function that is generalized to accept non Go source files, this would enforce more consistency around setting a breakpoint on a function.

Thoughts? I'm really not tied to either solution.